apis/server/router.go

```diff
@@ -17,50 +17,53 @@ import (
+	r.Path(versionMatcher + "/containers/{name:.*}/unpause").Methods(http.MethodPost).Handler(s.filter(s.unpauseContainer))
+	r.Path(versionMatcher + "/containers/{name:.*}/update").Methods(http.MethodPost).Handler(s.filter(s.updateContainer))
 
 	// image
-	r.Path("/images/create").Methods(http.MethodPost).Handler(s.filter(s.pullImage))
-	r.Path("/images/search").Methods(http.MethodGet).Handler(s.filter(s.searchImages))
-	r.Path("/images/json").Methods(http.MethodGet).Handler(s.filter(s.listImages))
-	r.Path("/images/{name:.*}").Methods(http.MethodDelete).Handler(s.filter(s.removeImage))
-	r.Path("/images/{name:.*}/json").Methods(http.MethodGet).Handler(s.filter(s.getImage))
+	r.Path(versionMatcher + "/images/create").Methods(http.MethodPost).Handler(s.filter(s.pullImage))
+	r.Path(versionMatcher + "/images/search").Methods(http.MethodGet).Handler(s.filter(s.searchImages))
+	r.Path(versionMatcher + "/images/json").Methods(http.MethodGet).Handler(s.filter(s.listImages))
+	r.Path(versionMatcher + "/images/{name:.*}").Methods(http.MethodDelete).Handler(s.filter(s.removeImage))
+	r.Path(versionMatcher + "/images/{name:.*}/json").Methods(http.MethodGet).Handler(s.filter(s.getImage))
 
 	// volume
-	r.Path("/volumes").Methods(http.MethodGet).Handler(s.filter(s.listVolume))
-	r.Path("/volumes/create").Methods(http.MethodPost).Handler(s.filter(s.createVolume))
-	r.Path("/volumes/{name:.*}").Methods(http.MethodGet).Handler(s.filter(s.getVolume))
-	r.Path("/volumes/{name:.*}").Methods(http.MethodDelete).Handler(s.filter(s.removeVolume))
+	r.Path(versionMatcher + "/volumes").Methods(http.MethodGet).Handler(s.filter(s.listVolume))
+	r.Path(versionMatcher + "/volumes/create").Methods(http.MethodPost).Handler(s.filter(s.createVolume))
+	r.Path(versionMatcher + "/volumes/{name:.*}").Methods(http.MethodGet).Handler(s.filter(s.getVolume))
+	r.Path(versionMatcher + "/volumes/{name:.*}").Methods(http.MethodDelete).Handler(s.filter(s.removeVolume))
 
 	// metrics
-	r.Path("/metrics").Methods(http.MethodGet).Handler(prometheus.Handler())
+	r.Path(versionMatcher + "/metrics").Methods(http.MethodGet).Handler(prometheus.Handler())
 
 	// network
-	r.Path("/networks").Methods(http.MethodGet).Handler(s.filter(s.listNetwork))
-	r.Path("/networks/create").Methods(http.MethodPost).Handler(s.filter(s.createNetwork))
-	r.Path("/networks/{name:.*}").Methods(http.MethodGet).Handler(s.filter(s.getNetwork))
-	r.Path("/networks/{name:.*}").Methods(http.MethodDelete).Handler(s.filter(s.deleteNetwork))
+	r.Path(versionMatcher + "/networks").Methods(http.MethodGet).Handler(s.filter(s.listNetwork))
+	r.Path(versionMatcher + "/networks/create").Methods(http.MethodPost).Handler(s.filter(s.createNetwork))
+	r.Path(versionMatcher + "/networks/{name:.*}").Methods(http.MethodGet).Handler(s.filter(s.getNetwork))
+	r.Path(versionMatcher + "/networks/{name:.*}").Methods(http.MethodDelete).Handler(s.filter(s.deleteNetwork))
```

Comment:
I do not think the above code is reasonable, so much redundant code. I will try to make it better.

Reply:
Yes, we may need a router wapper.